src/Components/Users/ManageUsers.tsx

```diff
@@ -154,43 +135,45 @@ export default function ManageUsers() {
       setWeeklyHoursError("Value should be between 0 and 168");
       return;
     }
-    const res = await dispatch(
-      partialUpdateUser(username, {
-        weekly_working_hours: weeklyHours,
-      })
-    );
-
-    if (res?.data) {
+    const { res, data } = await request(routes.partialUpdateUser, {
+      pathParams: { username },
+      body: { weekly_working_hours: weeklyHours },
+    });
+    if (res && res.status === 200 && data) {
       Notification.Success({
         msg: "Working hours updated successfully",
       });
       setExpandWorkingHours(false);
       setSelectedUser(null);
     } else {
       Notification.Error({
-        msg: "Error while updating working hours: " + (res.data.detail || ""),
+        msg: "Error while updating working hours: " + (data?.detail || ""),
       });
     }
     setWeeklyHours(0);
     setWeeklyHoursError("");
-    fetchData({ aborted: false });
+    await fetchDistrict();
+    await refetchUserList();
   };
 
   const handleSubmit = async () => {
     const username = userData.username;
-    const res = await dispatch(deleteUser(username));
+    const { res, data } = await request(routes.deleteUser, {
+      body: { username },
+    });
     if (res?.status === 204) {
       Notification.Success({
         msg: "User deleted successfully",
       });
     } else {
       Notification.Error({
-        msg: "Error while deleting User: " + (res?.data?.detail || ""),
+        msg: "Error while deleting User: " + (data?.detail || ""),
```

Comment:
Same, is this working? Error won't be present in data, it'll be in error instead

```suggestion
        msg: "Error while deleting User: " + (error?.detail || ""),
```